### PR TITLE
Expose identity token again

### PR DESCRIPTION
**Pull request by HenriBeck**
#62

### CODE REVIEW

File: packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart [hunk 1]
```diff
@@ -197,7 +197,7 @@ class SignInWithApple {
             .join(' '),
         // Request `code`, which is also what `ASAuthorizationAppleIDCredential.authorizationCode` contains.
         // So the same handling can be used for Apple and 3rd party platforms
-        'response_type': 'code',
+        'response_type': 'code+id_token',
```
Comment (HenriBeck) [inline packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart:200]: @tp Do you think it makes sense to configure this from the outside?

Comment (HenriBeck) [reply packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart:200]: FYI: I added the `id_token` response param according to https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_js/incorporating_sign_in_with_apple_into_other_platforms#3332113

The `code+id_token` has been taken from https://forums.developer.apple.com/thread/118209. I still need to validate this

Comment (HenriBeck) [reply packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart]: Update: I tried the android code and it seems to be `code id_token` in the query param so I adjusted the code for this.

**Comment by tp** [comment]
@HenriBeck Do you have any insight on how this works? It says that the identity token is basically the JWT token, which is (to the best of my knowledge) also what you get back when you ask Apple with the `code`. So why would one prefer to use the identity token instead?

(Wondering whether this would also simplify anything for us.)

File: packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart (hunk 1 above)
Comment (tp) [reply packages/sign_in_with_apple/lib/src/sign_in_with_apple.dart]: 👍

**Pull request merged**